scanomatic/data_processing/phenotyper.py

```diff
@@ -322,73 +328,62 @@ def LoadFromState(cls, directory_path: str) -> "Phenotyper":
         smooth_growth_data = unpickle_with_unpickler(
             np.load,
             os.path.join(directory_path, _p.phenotypes_input_smooth),
+            allow_pickle=True,
         )
 
         try:
-            extraction_params = unpickle_with_unpickler(
-                np.load,
-                os.path.join(directory_path, _p.phenotypes_extraction_params),
-            )
+            extraction_params = jsonizer.load(os.path.join(
+                directory_path,
+                _p.phenotypes_extraction_params,
+            ))
         except IOError:
             phenotyper._logger.warning(
                 "Could not find stored extraction parameters, assuming defaults were used",  # noqa: E501
             )
         else:
-            if extraction_params.size > 0:
-                if extraction_params.size == 3:
-                    (
-                        median_filt_size,
-                        gauss_sigma,
-                        linear_reg_size,
-                    ) = extraction_params
-                elif extraction_params.size == 4:
-                    (
-                        median_filt_size,
-                        gauss_sigma,
-                        linear_reg_size,
-                        inclusion_name,
-                    ) = extraction_params
-                    if inclusion_name is None:
-                        inclusion_name = 'Trusted'
+            try:
+                phenotype_inclusion = extraction_params.get(
+                    "phenotype_inclusion",
+                )
+                if phenotype_inclusion is not None:
```

Comment:
perhaps rewrite with `:=`

Reply:
The thought struck me, but everytime I've tried being clever with the `:=` and `is None` it has backfired on some technicality. I can have another look, but don't hold your breath.

Reply:
maybe missing parenthesis

Reply:
fixed